test/sentry_browser_test.dart

```diff
@@ -0,0 +1,425 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```

Comment:
It looks like a lot of the test code is identical with the `io` version. Would it be possible to have shared test code configurable via a `typedef SentryClientFactory` that provides either `io` or `browser` implementation of the client??

Reply:
The conditional import made it really simple to do, I moved all common and configurable tests under the `test_utils.dart` file using a single function 
```dart
runTest({Codec<List<int>, List<int>> gzip, bool isWeb = false})
```

Also move `Event` test under `event_test.dart` file.